### ISSUE TITLE
[Bug]: An error occurred when quantifying image-text dataset, Qwen3-VL-4B 'Tensor' object has no attribute 'get'

### ISSUE THREAD

**Issue by justzhanghong** [vllm-project/llm-compressor#2000]
### ⚙️ Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Operating System: `Linux-5.4.0-125-generic-x86_64-with-glibc2.35`
Python Version: `3.12.11 (main, Jun  4 2025, 08:56:18) [GCC 11.4.0]`
llm-compressor Version: `0.8.1`
compressed-tensors Version: `0.12.2`
transformers Version: `4.57.1`
torch Version: `2.8.0`
CUDA Devices: `['NVIDIA A100-SXM4-80GB']`
AMD Devices: `None`
```

</details>

### 🐛 Describe the bug

If a text dataset is used, quantization can be performed correctly.
An error occurred when quantifying image-text dataset: https://github.com/vllm-project/llm-compressor/issues/1941

### 🛠️ Steps to reproduce

import os
import sys
import llmcompressor
import transformers

print("----" * 20)
print(llmcompressor.__file__)
print("----" * 20)
print(transformers.__file__)
print("----" * 20)

import base64
from io import BytesIO

import torch
torch._dynamo.config.suppress_errors = True

from datasets import load_dataset
from qwen_vl_utils import process_vision_info
from transformers import AutoProcessor, Qwen3VLForConditionalGeneration

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.utils import dispatch_for_generation

# Load model.
model_id = "./Qwen/Qwen3-VL-4B-Instruct"
model = Qwen3VLForConditionalGeneration.from_pretrained(model_id, torch_dtype="auto", device_map="auto")
processor = AutoProcessor.from_pretrained(model_id, trust_remote_code=True)

# Oneshot arguments
DATASET_ID = "./flikr30k"
DATASET_SPLIT = "test[:12]"
NUM_CALIBRATION_SAMPLES = 12
MAX_SEQUENCE_LENGTH = 2048

# Load dataset and preprocess.
ds = load_dataset(DATASET_ID, split=DATASET_SPLIT)
ds = ds.shuffle(seed=42)

# Apply chat template and tokenize inputs.
def preprocess_and_tokenize(example):
    # preprocess
    buffered = BytesIO()
    example["image"].save(buffered, format="PNG")
    encoded_image = base64.b64encode(buffered.getvalue())
    encoded_image_text = encoded_image.decode("utf-8")
    base64_qwen = f"data:image;base64,{encoded_image_text}"
    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": base64_qwen},
                {"type": "text", "text": "What does the image show?"},
            ],
        }
    ]
    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    image_inputs, video_inputs = process_vision_info(messages)

    # tokenize
    return processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=False,
        max_length=MAX_SEQUENCE_LENGTH,
        truncation=True,
    )

ds = ds.map(preprocess_and_tokenize, remove_columns=ds.column_names)

# Define a oneshot data collator for multimodal inputs.
def data_collator(batch):
    assert len(batch) == 1
    return {key: torch.tensor(value) for key, value in batch[0].items()}

# Recipe
recipe = [
    GPTQModifier(
        targets="Linear",
        scheme="W4A16",
        ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
    ),
]

# Perform oneshot
oneshot(
    model=model,
    tokenizer=model_id,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    trust_remote_code_model=True,
    data_collator=data_collator,
    sequential_targets=["Qwen3_VLDecoderLayer"],
)

# Confirm generations of the quantized model look sane.
print("========== SAMPLE GENERATION ==============")
dispatch_for_generation(model)
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "http://images.cocodataset.org/train2017/000000231895.jpg",
            },
            {"type": "text", "text": "Please describe the animal in this image\n"},
        ],
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[prompt],
    images=image_inputs,
    videos=video_inputs,
    padding=False,
    max_length=MAX_SEQUENCE_LENGTH,
    truncation=True,
    return_tensors="pt",
).to(model.device)
output = model.generate(**inputs, max_new_tokens=100)
print(processor.decode(output[0], skip_special_tokens=True))
print("==========================================")

# Save to disk compressed.
SAVE_DIR = model_id.rstrip("/").split("/")[-1] + "-W4A16-G128"
model.save_pretrained(SAVE_DIR, save_compressed=True)
processor.save_pretrained(SAVE_DIR)

**Comment by kylesayrs** [comment]
As noted in the previous issue, calibrating with a text dataset is a valid workaround.

I suspect that the `get` attribute being described might be here?
https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_vl/modeling_qwen3_vl.py#L1531

I'm currently looking into other techniques to break up subgraphs such as https://github.com/vllm-project/llm-compressor/pull/1999

**Comment by justzhanghong** [comment]
I plan to quantize the visual branch of Qwen3-VL as well, so I need to use a image-text dataset.

**Comment by reckness** [comment]
Hi, you can solve it through the #1930

**Comment by justzhanghong** [comment]
> Hi, you can solve it through the [#1930](https://github.com/vllm-project/llm-compressor/pull/1930)

I tried this branch, but I'm still getting the same error.

**Comment by reckness** [comment]
You can try it; I can run it under the #1930 branch.
```python
import base64
from io import BytesIO

import torch
from datasets import load_dataset
from qwen_vl_utils import process_vision_info
from transformers import AutoProcessor, Qwen3VLForConditionalGeneration

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier 
from llmcompressor.modifiers.awq import AWQModifier
from llmcompressor.utils import dispatch_for_generation
from llmcompressor.modeling import replace_modules_for_calibration

# Load model.
model_id = "Qwen3-VL-8B-Instruct"
model = Qwen3VLForConditionalGeneration.from_pretrained(model_id, device_map=None,torch_dtype="auto",local_files_only=True)
processor = AutoProcessor.from_pretrained(model_id, trust_remote_code=True)
# for name, module in model.named_modules():
#     print(name, ":", module)

# Oneshot arguments
DATASET_ID = "lmms-lab/flickr30k"
DATASET_SPLIT = "test[:512]"
NUM_CALIBRATION_SAMPLES = 512
MAX_SEQUENCE_LENGTH = 2048

# Load dataset and preprocess.
ds = load_dataset(DATASET_ID, split=DATASET_SPLIT)
ds = ds.shuffle(seed=42)


# Apply chat template and tokenize inputs.
def preprocess_and_tokenize(example):
    # preprocess
    buffered = BytesIO()
    example["image"].save(buffered, format="PNG")
    encoded_image = base64.b64encode(buffered.getvalue())
    encoded_image_text = encoded_image.decode("utf-8")
    base64_qwen = f"data:image;base64,{encoded_image_text}"
    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": base64_qwen},
                {"type": "text", "text": "What does the image show?"},
            ],
        }
    ]
    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    image_inputs, video_inputs = process_vision_info(messages)

    # tokenize
    return processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=False,
        max_length=MAX_SEQUENCE_LENGTH,
        truncation=True,
    )


ds = ds.map(preprocess_and_tokenize, remove_columns=ds.column_names)


# Define a oneshot data collator for multimodal inputs.
def data_collator(batch):
    assert len(batch) == 1
    return {key: torch.tensor(value) for key, value in batch[0].items()}


# Recipe
# recipe = [
#     GPTQModifier(
#         targets="Linear",
#         scheme="W4A16",
#         ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
#     ),
# ]
recipe = [
    AWQModifier(
        ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
        duo_scaling=False,
        config_groups={
             "group_0": {
                 "targets":{"Linear"},
                 "input_activations" : None,
                 "output_activations": None,
              "weights": {
                  "num_bits": 4,
                  "type": "int",
                  "symmetric": True,
                  "strategy": "group",
                  "group_size": 32,
                  "observer": "mse"
                },
                },       
         }
    ),
]

# Perform oneshot
oneshot(
    model=model,
    tokenizer=model_id,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    trust_remote_code_model=True,
    data_collator=data_collator,
    sequential_targets=["Qwen3VLTextDecoderLayer"],
)

# Confirm generations of the quantized model look sane.
print("========== SAMPLE GENERATION ==============")
dispatch_for_generation(model)
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "http://images.cocodataset.org/train2017/000000231895.jpg",
            },
            {"type": "text", "text": "Please describe the animal in this image\n"},
        ],
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[prompt],
    images=image_inputs,
    videos=video_inputs,
    padding=False,
    max_length=MAX_SEQUENCE_LENGTH,
    truncation=True,
    return_tensors="pt",
).to(model.device)
output = model.generate(**inputs, max_new_tokens=100)
print(processor.decode(output[0], skip_special_tokens=True))
print("==========================================")


# Save to disk compressed.
SAVE_DIR = model_id.rstrip("/").split("/")[-1] + "-W4A16-G129"
model.save_pretrained(SAVE_DIR, save_compressed=True)
processor.save_pretrained(SAVE_DIR)
```

**Comment by brian-dellabetta** [comment]
@justzhanghong @reckness , just FYI, the underlying issue is resolved on transformers, I believe from this PR -- https://github.com/huggingface/transformers/pull/41536

If you are on transformers main (and llm-compressor main) @reckness 's example just above should work. After some discussion, we decided not to pursue #1930 out of correctness concerns.

**Comment by justzhanghong** [comment]
@brian-dellabetta @reckness Thank you, it ran successfully. 
```python
import base64
import argparse
from io import BytesIO

import torch
from datasets import load_dataset
from qwen_vl_utils import process_vision_info
from transformers import AutoProcessor, Qwen3VLForConditionalGeneration

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier 
from llmcompressor.modifiers.awq import AWQModifier
from llmcompressor.utils import dispatch_for_generation
from llmcompressor.modeling import replace_modules_for_calibration

# Load model.
model_id = "Qwen/Qwen3-VL-4B-Instruct"
model = Qwen3VLForConditionalGeneration.from_pretrained(model_id, device_map=None,torch_dtype="auto",local_files_only=True)
processor = AutoProcessor.from_pretrained(model_id, trust_remote_code=True)
# for name, module in model.named_modules():
#     print(name, ":", module)

# Oneshot arguments
DATASET_ID = "flikr30k"
DATASET_SPLIT = "test[:12]"
NUM_CALIBRATION_SAMPLES = 12
MAX_SEQUENCE_LENGTH = 2048

# Load dataset and preprocess.
ds = load_dataset(DATASET_ID, split=DATASET_SPLIT)
ds = ds.shuffle(seed=42)


# Apply chat template and tokenize inputs.
def preprocess_and_tokenize(example):
    # preprocess
    buffered = BytesIO()
    example["image"].save(buffered, format="PNG")
    encoded_image = base64.b64encode(buffered.getvalue())
    encoded_image_text = encoded_image.decode("utf-8")
    base64_qwen = f"data:image;base64,{encoded_image_text}"
    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": base64_qwen},
                {"type": "text", "text": "What does the image show?"},
            ],
        }
    ]
    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    image_inputs, video_inputs = process_vision_info(messages)

    # tokenize
    return processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=False,
        max_length=MAX_SEQUENCE_LENGTH,
        truncation=True,
    )


ds = ds.map(preprocess_and_tokenize, remove_columns=ds.column_names)


# Define a oneshot data collator for multimodal inputs.
def data_collator(batch):
    assert len(batch) == 1
    return {key: torch.tensor(value) for key, value in batch[0].items()}

# Recipe
# recipe = [
#     GPTQModifier(
#         targets="Linear",
#         scheme="W4A16",
#         ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
#     ),
# ]

# recipe = [
#     AWQModifier(
#         ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
#         duo_scaling=False,
#         config_groups={
#              "group_0": {
#                  "targets":{"Linear"},
#                  "input_activations" : None,
#                  "output_activations": None,
#               "weights": {
#                   "num_bits": 4,
#                   "type": "int",
#                   "symmetric": True,
#                   "strategy": "group",
#                   "group_size": 32,
#                   "observer": "mse"
#                 },
#               },       
#          }
#     ),
# ]

parser = argparse.ArgumentParser(description="vision block range")
parser.add_argument('--max_id', type=int, default=-1, help='block id')
args = parser.parse_args()

vit_ignore = list()
for i in range(args.max_id):
    vit_ignore.append('"re:model.visual.blocks.{}.attn.*"'.format(i))
    vit_ignore.append('"re:model.visual.blocks.{}.mlp.*"'.format(i))
vit_ignore_str = ",".join(vit_ignore)

recipe = """
quant_stage:
    quant_modifiers:
        GPTQModifier:
            ignore: [
                        "lm_head",
                        {}
                        "re:model.visual.deepstack_merger_list.*"
                    ]
            config_groups:
                group_0:
                    weights:
                        num_bits: 4
                        type: int
                        strategy: channel
                        dynamic: false
                        symmetric: true
                    input_activations:
                        num_bits: 8
                        type: int
                        strategy: token
                        dynamic: true
                        symmetric: true
                    targets: ["re:model.language_model.layers.*self_attn.q_proj.*", "re:model.language_model.layers.*self_attn.k_proj.*", "re:model.language_model.layers.*self_attn.v_proj.*", "re:model.language_model.layers.*self_attn.o_proj.*", "re:model.language_model.layers.*mlp.gate_proj.*", "re:model.language_model.layers.*mlp.up_proj.*", "re:model.language_model.layers.*mlp.down_proj.*"]
                group_1:
                    weights:
                        num_bits: 8
                        type: int
                        strategy: tensor
                        dynamic: false
                        symmetric: true
                    input_activations:
                        num_bits: 8
                        type: int
                        strategy: tensor
                        dynamic: false
                        symmetric: true
                    targets: ["re:model.visual.blocks.*attn.qkv.*", "re:model.visual.blocks.*attn.proj.*", "re:model.visual.blocks.*mlp.linear_fc1.*", "re:model.visual.blocks.*mlp.linear_fc2.*"]
""".format(vit_ignore_str + "," if vit_ignore_str else "")



# Perform oneshot
oneshot(
    model=model,
    tokenizer=model_id,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    trust_remote_code_model=True,
    data_collator=data_collator,
    sequential_targets=["Qwen3VLTextDecoderLayer"],
)

# Confirm generations of the quantized model look sane.
print("========== SAMPLE GENERATION ==============")
dispatch_for_generation(model)
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "http://images.cocodataset.org/train2017/000000231895.jpg",
            },
            {"type": "text", "text": "Please describe the animal in this image\n"},
        ],
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[prompt],
    images=image_inputs,
    videos=video_inputs,
    padding=False,
    max_length=MAX_SEQUENCE_LENGTH,
    truncation=True,
    return_tensors="pt",
).to(model.device)
output = model.generate(**inputs, max_new_tokens=100)
print(processor.decode(output[0], skip_special_tokens=True))
print("==========================================")


# Save to disk compressed.
SAVE_DIR = model_id.rstrip("/").split("/")[-1] + "-W8A8_ViT-W4A8-LLM"
model.save_pretrained(SAVE_DIR, save_compressed=True)
processor.save_pretrained(SAVE_DIR)
```
